sdk/storage/Azure.Storage.Common/src/IPRange.cs

```diff
@@ -3,69 +3,122 @@
 // license information.
 
 using System;
+using System.ComponentModel;
 using System.Net;
 
 namespace Azure.Storage.Common
 {
     /// <summary>
-    /// Represents a SAS IP range's start IP and (optionally) end IP.
+    /// Represents a range of allowed IP addresses for constructing a Shared
+    /// Access Signature.
     /// </summary>
-    public struct IPRange : IEquatable<IPRange>
+    public readonly struct IPRange : IEquatable<IPRange>
     {
         /// <summary>
-        /// Not specified if equal to IPAddress.None
+        /// Gets the start of the IP range.  Not specified if equal to null or
+        /// <see cref="IPAddress.None"/>.
         /// </summary>
-        public IPAddress Start { get; set; }
+        public IPAddress Start { get; }
 
         /// <summary>
-        /// Not specified if equal to IPAddress.None
+        /// Gets the optional end of the IP range.  Not specified if equal to
+        /// null or <see cref="IPAddress.None"/>.
         /// </summary>
-        public IPAddress End { get; set; }
+        public IPAddress End { get; }
 
         /// <summary>
-        /// String returns a string representation of an IPRange.
+        /// Creates a new <see cref="IPRange"/>.
         /// </summary>
-        /// <returns></returns>
-        public override string ToString()
+        /// <param name="start">
+        /// The range's start <see cref="IPAddress"/>.
+        /// </param>
+        /// <param name="end">
+        /// The range's optional end <see cref="IPAddress"/>.
+        /// </param>
+        public IPRange(IPAddress start, IPAddress end = null)
         {
-            var start = this.Start ?? IPAddress.None;
-            var end = this.End ?? IPAddress.None;
-
-            return
-                start == IPAddress.None
-                ? String.Empty
-                : end == IPAddress.None
-                ? start.ToString()
-                : start.ToString() + "-" + end.ToString();
+            this.Start = start ?? IPAddress.None;
+            this.End = end ?? IPAddress.None;
         }
+
+        /// <summary>
+        /// Check if an <see cref="IPAddress"/> was not provided.
+        /// </summary>
+        /// <param name="address">The address to check.</param>
+        /// <returns>True if it's empty, false otherwise.</returns>
+        private static bool IsEmpty(IPAddress address) =>
+            address == null || address == IPAddress.None;
+
+        /// <summary>
+        /// Creates a string representation of an <see cref="IPRange"/>.
+        /// </summary>
+        /// <returns>
+        /// A string representation of an <see cref="IPRange"/>.
+        /// </returns>
+        public override string ToString() =>
```

Comment:
NIT: Could we make this into clear `if - else` statements? it would be easier to read

Reply:
Same everywhere where you do that

Reply:
The Storage team uses these a bunch and I'm hesitant to make that change (and other valid style points you've called out) globally as part of this work.